algorithms/C/BubbleSort/bubblesort.c

```diff
@@ -1,38 +1,71 @@
-#include<stdio.h>
-
-int main()
-{
-int n,i,j,temp;
-printf("Enter number of elements in array\n");
-scanf("%d",&n);
-int A[n];
-printf("Array before sorting\n");
-
-for(i=0;i<n;i++)
-	{
-	A[i]=rand()%1000;
-	printf("%d ",A[i]);	
-	}
-printf("\n");
-
-for(i=0;i<n;i++)
-	{
-	for(j=n;j>i;j--)
-		{
-		  if(A[j]<A[j-1])
-			{
-			temp=A[j];
-			A[i]=A[i-1];
-			A[i-1]=temp;
-			
-			}	
-		}	
-	}
-printf("Array after sorting\n");
-
-for(i=0;i<n;i++) printf("%d ",A[i]);
-
-printf("\n");
-	
-return 0;
-}
\ No newline at end of file
+#include <stdio.h>
+
+void bubbleSort(int array[], int size, int order){
+  int aux = 0;
+
+  if(order == 1){
+    for(int i=0; i < size-1; i++){
+      int flag = 0;
+
+      for(int j=0; j<size-1-i; j++){
+        if(array[j] > array[j+1]){
+          aux = array[j]; 
+          array[j] = array[j+1]; 
+          array[j+1] = aux; 
+          flag = 1;
```

Comment:
This is exactly the same as line 31-34, move it to a util function

Reply:
Done! Check pls :)